### ISSUE TITLE
Profile Picker "Ask On Startup" checkbox broken

### ISSUE THREAD

**Issue by ghost** [brave/brave-browser#41194]
### Description

**When I disable show on startup it always shows**

> When I disable show on startup it always shows

### Steps to reproduce

1. Open the Brave browser, click on "Customize and control brave" in the top right
2. More tools > Add new profile
3. Disable the show on startup option. (checkbox disabled)
4. exit the browser and start it again

### Actual result

![1](https://github.com/user-attachments/assets/b618d679-aff4-4689-8825-f1a46261638b)

### Expected result

If the option is disabled, Brave must be in the latest profile when launched.

### Reproduces how often

Easily reproduced

### Brave version (brave://version info)

```
Brave: 1.70.117 Chromium: 129.0.6668.59 (Official Build) (64-bit) 
Revision:	5d4da8332eb061f79bb47ec5b0bf595600fdcd55
OS: Linux
```

> [!NOTE]
> Also version `nightly`  has the same issue

```
Brave: 1.72.26 Chromium: 129.0.6668.59 (Official Build) nightly (64-bit) 
Revision: 4580158f4966388aa15f701b5fa01cbe22846cc5
OS: Linux
```

### Channel information

- [X] release (stable)
- [X] nightly
- [ ] beta

### Reproducibility

- [X] with Brave Shields disabled
- [X] with Brave Rewards disabled
- [X] in the latest version of Chrome

**Comment by neonig** [comment]
Are there any developments?

**Comment by avizitRX** [comment]
Facing this same issue on Ubuntu

**Comment by A-t-s-u-s-h-i** [comment]
+1 from another user. And I can reproduce the same issue on MacOS 14.6.1/Brave 1.70.119. Didn't reproduce on Brave 1.69.x.

**Comment by bsclifton** [comment]
cc: @emerick @cdesouza-chromium - may have been a regression w/ Chromium 129?

**Comment by bsclifton** [comment]
Regression - tracked upstream with https://issues.chromium.org/issues/365405769

Thanks @cdesouza-chromium for the fix 😄

**Comment by LaurenWags** [comment]
FYI, we thought the minor bump from https://github.com/brave/brave-browser/issues/41338 would address this, but it did not:


https://github.com/user-attachments/assets/01302cbe-c381-4029-8744-d3ac20f44ba3

**Comment by LaurenWags** [comment]
Requires 1.70.124 or higher for verification 👍🏻

**Comment by DylanManiatakes** [comment]
Problem still happening on MacOS on latest update

**Comment by kjozwiak** [comment]
> Problem still happening on MacOS on latest update

@DylanManiatakes the above will go out via the next `1.70.x` release which will most likely happen sometime this week once we get the next minor `C129` 👍 Believe we anticipated that the upstream fix would have addressed the above as per https://github.com/brave/brave-browser/issues/41194#issuecomment-2386633502. However, that wasn't the case so we needed to uplift/cherry-pick our own fix into `1.71.x` & `1.70.x` via https://github.com/brave/brave-browser/issues/41194#event-1510398627.

**Comment by GeetaSarvadnya** [comment]
Verification **PASSED** on

```
Brave | 1.70.126 Chromium: 129.0.6668.100 (Official Build) (64-bit)
-- | --
Revision | 35f34f21e597f5f30e9e206b0354cc137df1f2a3
OS | Windows 10 Version 22H2 (Build 19045.4894)

```
- Confirmed that `Profile manager` window is not shown after browser restart
- Confirmed that latest profile/old profile is shown when browser restarted
- Confirmed that `Show on startup` remained unchecked via profile manager after browser restart 

![Image](https://github.com/user-attachments/assets/b405aad0-40e4-4b67-a42f-bc9384436a4f)
